etna/datasets/internal_datasets.py

```diff
@@ -239,12 +259,112 @@ def get_m4_dataset(dataset_dir: Path, dataset_freq: str) -> None:
     )
 
 
+def get_m3_dataset(dataset_dir: Path, dataset_freq: str) -> None:
+    """
+    Download and save M3 dataset in different frequency modes.
+
+    The M3 dataset is a collection of 3,003 time series used for the third edition of the Makridakis forecasting
+    Competition. The M3 dataset consists of time series of yearly, quarterly, monthly and other data.
```

Comment:
- Let's explain what is other data. That it doesn't have any particular frequency but we assume it is a quarterly data.
- Let's give a clarification about the timestamps in `df_exog`. Don't forget to mention situation with `other`.